compose/service.py

```diff
@@ -399,6 +401,12 @@ def _get_container_create_options(self, override_options, one_off=False):
             self.containers(stopped=True, one_off=one_off),
             one_off)
 
+        if 'labels' in container_options:
+            labels_dict = ['io.docker.compose:' + label for label in container_options['labels']]
```

Comment:
Agreed that the constraints even though stored as labels should not all share the same namespace. It must be known what is a constraint, what is an identification label, what is a capability label, etc. 

An operator should be able to say at `docker-compose run` time to ignore all previous constraints and use a new set. This should not require the operator to overwrite each constraint separately. Allowing for this separation of concern is absolutely critical. It must be possible to tie applications to infrastructure without storing this glue in the application itself.

So instead of separating compose and runtime labels with a namespace, an alternative could be a `--remove-labels` option which allows for wildcards:

```yaml
web:
    labels:
         constraints:
            storage: ssd
            memory: 2
         generic:
            hello: world
```

To overwrite all constraints:

`[...] --remove-labels 'constraints.*' --label constraints.memory=4g`